cfme/tests/configure/test_default_views_infra.py

```diff
@@ -50,6 +52,26 @@ def set_and_test_default_view(group_name, view, page):
     assert tb.is_active(view), "{} view setting failed".format(view)
     DefaultView.set_default_view(group_name, old_default)
 
+
+def check_vm_visibility():
+    navigate_to(Vm, 'All')
+    app = get_or_create_current_appliance()
+    view = app.browser.widgetastic.create_view(ExploreVmView)
+    value = view.accordions.VmAndTemplates.tree.read_contents()
+    tree_root = value[0]
+    tree_provider = value[1][0][0]
+    tree_datacenter = value[1][0][1][0][0]
+    length = len(value[1][0][1][0][1])
+    tree_vm = value[1][0][1][0][1][length - 1]
```

Comment:
These numbers look a bit arbitrary, maybe some explanations about why those?

Reply:
**Required**

Use `-1` index to get the last item in tree_vm, don't calculate the length.

Expand your comments, include an example of the data structure you're trying to parse with these indexes.

If vm visibility is disabled, these index trees will cause index errors, because some of the elements won't exist.